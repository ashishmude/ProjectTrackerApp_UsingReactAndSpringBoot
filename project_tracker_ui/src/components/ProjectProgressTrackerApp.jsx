import React, { Component } from "react";
import { withStyles } from "@material-ui/core/styles";
import {
  AppBar,
  Toolbar,
  Typography,
  IconButton,
  makeStyles,
} from "@material-ui/core";
import ListCheckpointsComponent from "./ListCheckpointsComponent";
import ProjectService from "../service/ProjectService";


const styles = (theme) => ({
  titleBar: {
      justifyContent: "center",
      display: "flex",
    },
});

class ProjectProgressTrackerApp extends Component {
  constructor(props) {
      super(props);
      this.state = {
        project: {}
      };
      this.refreshProjectDetails = this.refreshProjectDetails.bind(this);
    }
    componentDidMount() {
      this.refreshProjectDetails();
    }
    refreshProjectDetails() {
     ProjectService.retrieveProjectDetails()
       .then(response => {
           console.log(response);
           this.setState({ project: response.data });
       })
    }

  render() {
    const  { project } = this.state;
    const { classes } = this.props;
    return (
      <>
        <AppBar position="static">
          <Toolbar className={classes.titleBar}>
            <IconButton
              edge="start"
              color="inherit"
              aria-label="menu"
            ></IconButton>
            <Typography variant="h6">{ project.name } : { project.description }</Typography>
          </Toolbar>
        </AppBar>
        <ListCheckpointsComponent />
       </>
    );
  }
};

export default withStyles(styles)(ProjectProgressTrackerApp);
