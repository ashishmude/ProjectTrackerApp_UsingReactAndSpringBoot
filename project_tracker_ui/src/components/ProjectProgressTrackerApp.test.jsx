import React from 'react';
import { shallow } from 'enzyme';
import ProjectProgressTrackerApp from './ProjectProgressTrackerApp';


describe("ProjectProgressTrackerApp", () => {
  it("should render ProjectProgressTrackerApp", () => {
    const component = shallow(<ProjectProgressTrackerApp />);
    expect(component.getElements()).toMatchSnapshot();
  });
});
