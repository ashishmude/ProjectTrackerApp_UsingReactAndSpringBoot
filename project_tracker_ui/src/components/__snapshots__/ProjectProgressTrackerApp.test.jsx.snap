// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProjectProgressTrackerApp should render ProjectProgressTrackerApp 1`] = `
Array [
  <ProjectProgressTrackerApp
    classes={
      Object {
        "titleBar": "ProjectProgressTrackerApp-titleBar-1",
      }
    }
  />,
]
`;
